spec/porky_lib/file_service_spec.rb

```diff
@@ -400,4 +400,104 @@ def write_test_file(data)
       end.not_to raise_error
     end
   end
+
+  describe '#presigned_post_url' do
+    let(:s3_object) { instance_double(Aws::S3::Object) }
+
+    it 'returns presigned post url and fields' do
+      url, _file_name = file_service.presigned_post_url(bucket_name)
+      uri = URI.parse(url)
+      query_params = CGI.parse(uri.query)
+
+      expect(uri.scheme).to eq('https')
+      expect(uri.path).to include("/#{bucket_name}/")
+
+      expect(query_params["x-amz-server-side-encryption"]).to eq(["aws:kms"])
+    end
+
+    it 'uses file_name as key if provided' do
+      url, file_name = file_service.presigned_post_url(bucket_name, file_name: default_file_key)
+      uri = URI.parse(url)
+
+      expect(uri.path).to eq("/#{bucket_name}/#{default_file_key}")
+      expect(default_file_key).to eq(file_name)
+    end
+
+    it 'passes metadata if provided' do
```

Comment:
Maybe add one more test for the options like content-size-range, you know you are going to use that.

Reply:
As discussed, we are going to move the file size validation to the bucket as a policy. With that we'll also be enable to enforce content type validation and any other custom policy we may want to add in the future.